src/pudl/transform/vcerare.py

```diff
@@ -0,0 +1,330 @@
+    )
+    out_df = pd.concat([df[~bad_county_mask], grouped_df])
+    # Now lets remove the records for the other city with no values
+    bad_county_mask_2 = out_df["county_state_names"] == "bedford_city_virginia"
+    # Make sure there aren't any non-zero values
+    if not (out_df[bad_county_mask_2][cap_fac_cols] == 0).all().all():
+        raise AssertionError("Found non-zero values for bedford_city_virginia record.")
+    return out_df[~bad_county_mask_2]
+
+
+@asset(
+    io_manager_key="parquet_io_manager",
+    compute_kind="pandas",
+    op_tags={"memory-use": "high"},
+)
+def out_vcerare__hourly_available_capacity_factor(
+    raw_vcegen__lat_lon_fips: pd.DataFrame,
+    raw_vcerare__fixed_solar_pv_lat_upv: pd.DataFrame,
+    raw_vcerare__offshore_wind_power_140m: pd.DataFrame,
+    raw_vcerare__onshore_wind_power_100m: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform raw Vibrant Clean Energy renewable generation profiles.
+
+    Concatenates the solar and wind capacity factors into a single table and turns
+    the columns for each county or subregion into a single county_or_lake_name column.
+    """
+    logger.info("Transforming the hourly available capacity factor tables")
+    # Clean up the FIPS table
+    fips_df = _prep_lat_long_fips_df(raw_vcegen__lat_lon_fips)
+    # Apply the same transforms to all the capacity factor tables. This is slower
+    # than doing it to a concatinated table but less memory intensive because
+    # it doesn't need to process the ginormous table all at once.
+    raw_dict = {
+        "solar_pv": raw_vcerare__fixed_solar_pv_lat_upv,
+        "offshore_wind": raw_vcerare__offshore_wind_power_140m,
+        "onshore_wind": raw_vcerare__onshore_wind_power_100m,
+    }
+    clean_dict = {
+        df_name: _check_for_valid_counties(df, fips_df, df_name)
+        .pipe(_make_cap_fac_frac, df_name)
+        .pipe(_add_time_cols, df_name)
+        .pipe(_stack_cap_fac_df, df_name)
+        for df_name, df in raw_dict.items()
+    }
+    # Combine the data!
+    return (
+        _combine_all_cap_fac_dfs(clean_dict)
+        .pipe(_combine_city_county_records)
+        .pipe(_combine_cap_fac_with_fips_df, fips_df)
+    )
+
+
+@asset_check(
+    asset=out_vcerare__hourly_available_capacity_factor,
+    blocking=True,
+    description="Check that output table is as expected.",
+)
+def check_hourly_available_cap_fac_table(asset_df: pd.DataFrame):
+    """Check that the final output table is as expected."""
+    # Make sure the table is the expected length
+    if (length := len(asset_df)) != 136437000:
+        return AssetCheckResult(
+            passed=False,
+            description="Table unexpected length",
+            metadata={"table_length": length, "expected_length": 136437000},
+        )
+    # Make sure there are no null values
+    if asset_df.isna().any().any():
+        return AssetCheckResult(
+            passed=False, description="Found NA values when there should be none"
```

Comment:
Non-blocking: could be helpful to return the offending row PKs as part of the metadata

Reply:
I think it's easy enough to search the data for NA values I'm not too concerned about adding metadata for it.